crates/aptos/src/common/types.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fmt::Debug, str::FromStr};
+use structopt::StructOpt;
+use thiserror::Error;
+
+/// A common result to be returned to users
+pub type CliResult = Result<String, String>;
+
+/// A trait to keep common CLI command functions
+pub trait CliCommand {
+    /// Executes the given CLI command
+    fn execute(self) -> CliResult;
+}
+
+/// TODO: Re-evaluate these errors
+#[derive(Debug, Error)]
+pub enum Error {
```

Comment:
personally would just make this into the mod.rs or otherwise make per struct  / type files

Reply:
I expected it to grow, so I kept it separate for now.  Can always merge accordingly, I just didn't want it to become one of the really messy mod.rs files we have.